versions/1/elasticsearch/spec.yml

```diff
@@ -28,3 +28,13 @@ spec:
           pattern: '^.+\.json$'
           contentMediaType: "application/json"
           required: false
+    - description: Folder containing Elasticsearch Transforms
+      type: folder
+      name: transform
+      required: false
+      contents:
+        - description: An Elasticsearch transform file
+          # https://www.elastic.co/guide/en/elasticsearch/reference/current/transforms.html
+          type: file
+          contentMediaType: "application/json"
+          pattern: '^.+\.json$'
```

Comment:
Do you think we should apply any naming requirements? For example, to prepend a package name to the file. Is it possible that another package can override this transform?

Reply:
I'm not sure but I think the filename doesn't effect the created Transform at all.
As you can see in our integration (works locally) we have [two different transforms](https://github.com/elastic/integrations/pull/2930/files) in different directories with the same file name (`default.json`) and they don't override each other.
Probably @joshdover could explain better the exact meaning and implications the filenames and the containing directory names have.

Reply:
> endpoint security integration [transforms](https://github.com/elastic/endpoint-package/tree/master/package/endpoint/elasticsearch/transform) look the same.

From the spec perspective, all Endpoint security integration [transforms](https://github.com/elastic/endpoint-package/tree/master/package/endpoint/elasticsearch/transform) are illegal as they haven't been documented. This is the reason why you need to deal with it now :)

> I'm not sure but I think the filename doesn't effect the created Transform at all.
> As you can see in our integration (works locally) we have [two different transforms](https://github.com/elastic/integrations/pull/2930/files) in different directories with the same file name (default.json) and they don't override each other.

Let me elaborate here, what will happen if we have another package `foobar` with exact transforms and directories? When we install it, will it overwrite your transforms?

Reply:
cc @pzl

Reply:
> Let me elaborate here, what will happen if we have another package foobar with exact transforms and directories? When we install it, will it overwrite your transforms?

To my understanding the answer is no.
Seems like the id of the transform is a result of both the integration package name and the transform directory name.

![Screen Shot 2022-03-31 at 15 04 25](https://user-images.githubusercontent.com/63912106/161050912-696f049b-3fd3-4f8a-a3e2-94cea92fb1a7.png)

Additionally after having a conversation with @CohenIdo about the naming of the transform files I understood he was instructed to name them `default.json` so I'll change the spec to reflect this requirement.

Anyways I'm sure Josh will shed more light over this topic.

Reply:
@eyalkraft thanks for doing this, it's something I should have done some time back when my team originally added transforms.

@mtojek 

> I'm not sure but I think the filename doesn't effect the created Transform at all.
> Let me elaborate here, what will happen if we have another package `foobar` with exact transforms and directories? When we install it, will it overwrite your transforms?

Currently, when transforms are installed, the package name will be prepended (along with the directory path) to actual name of the transform.  Thus, it doesn't matter what the definition `.json` file is named.  So, as long as the packages themselves are named differently, I don't think there will be any collisions.  Here is what our transforms are named when installed.  (@pzl let me know if I have anything wrong here)

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/56395104/161048571-bdb1aa90-f907-4490-b0c0-ef707ce6ea36.png">

That being said, I am OK with any naming convention we want to enforce and we can update the names of files to match.

> We will enable the strict package validation soon and it will block the publishing of packages with undocumented (= illegal) features.

For clarity, adding this PR to the package spec will "document" the transform feature and will no longer make it illegal, correct?

Further, let us know when we can test against strict package enforcement so we can fix any other problems on our end and be prepared for when it is live.  I would ask that we can publish packages as is for the next couple weeks so that we can push our package with changes for the `8.2` release and in case we need to push anything additional to fix bugs.

Reply:
> That being said, I am OK with any naming convention we want to enforce and we can update the names of files to match.

That's the killer argument for this issue. I guess we can resolve the conversation. Thanks for jumping in, Kevin.

> For clarity, adding this PR to the package spec will "document" the transform feature and will no longer make it illegal, correct?

Yes, that's the idea.

> Further, let us know when we can test against strict package enforcement so we can fix any other problems on our end and be prepared for when it is live. I would ask that we can publish packages as is for the next couple weeks so that we can push our package with changes for the 8.2 release and in case we need to push anything additional to fix bugs.

@eyalkraft It would be great if we can copy Endpoint's transforms here as test cases to make sure they are aligned with new spec requirements. @kevinlog What would be the best source to copy from?

Reply:
@eyalkraft @mtojek 

Either of the definitions here are good: https://github.com/elastic/endpoint-package/tree/master/package/endpoint/elasticsearch/transform

Reply:
Added both